swarm/tests/public_api.rs

```diff
@@ -0,0 +1,64 @@
+use libp2p::core::transport::ListenerId;
+use libp2p::swarm::behaviour::FromSwarm;
+use libp2p::swarm::handler::DummyConnectionHandler;
+
+#[test]
+// test to break compilation everytime a variant changes,
+// forcing us to revisit each implementation
+fn swarm_event_variants() {
```

Comment:
@thomaseizinger following your suggestion on https://github.com/libp2p/rust-libp2p/pull/2867/files#r965651079 moved the test here